lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Addresses.hs

```diff
@@ -253,6 +257,83 @@ spec = describe "SHELLEY_ADDRESSES" $ do
+                    `shouldBe` p)
+                liftIO (length addr `shouldBe` i)
+
+        -- 1. create shelley wallet
+        wSrc <- fixtureWallet ctx
+        wDest <- emptyWalletWith ctx ("Wallet", "cardano-wallet", initPoolGap)
+
+        -- 2. list addresses of that wallet (for later comparison)
+        listAddresses @n ctx wSrc >>= verifyAddrs 50 10
+
+        -- 3. send a transaction from that wallet
+        addrs <- listAddresses @n ctx wDest
+        verifyAddrs 20 0 addrs
+        let destination = (addrs !! 0) ^. #id
+        let payload = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{2 * minUTxOValue},
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": "cardano-wallet"
+            }|]
+
+        request @(ApiTransaction n) ctx
+            (Link.createTransaction @'Shelley wSrc) Default payload
+            >>= flip verify
+                [ expectResponseCode HTTP.status202 ]
+
+        -- 4. wait for transaction to be inserted
+        eventually "Wallet balance = minUTxOValue" $ do
+            rb <- request @ApiWallet ctx
+                (Link.getWallet @'Shelley wDest) Default Empty
+            expectField
+                (#balance . #getApiT . #available)
+                (`shouldBe` Quantity (2 * minUTxOValue))
+                rb
+
+        -- 5. Check that change address is still there as unused
+        -- and the gap moved.
+        listAddresses @n ctx wSrc >>= verifyAddrs 51 11
+
+        -- 6. send another transaction from the destination wallet
+        listAddresses @n ctx wDest >>= verifyAddrs 21 1
+        addrs' <- listAddresses @n ctx wSrc
+        let destination' = (addrs' !! 22) ^. #id
+        let payload' = Json [json|{
+                "payments": [{
+                    "address": #{destination'},
+                    "amount": {
+                        "quantity": #{minUTxOValue},
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": "cardano-wallet"
+            }|]
+        currentSrcWallet <- getWallet ctx wSrc
+        let currentSrcWalletBalance = currentSrcWallet ^. (#balance . #getApiT . #available)
+        request @(ApiTransaction n) ctx
+            (Link.createTransaction @'Shelley wDest) Default payload'
+            >>= flip verify
+                [ expectResponseCode HTTP.status202 ]
+        eventually "Wallet balance = minUTxOValue + current" $ do
+            rb <- request @ApiWallet ctx
+                (Link.getWallet @'Shelley wSrc) Default Empty
+            expectField
+                (#balance . #getApiT . #available)
+                (`shouldBe` fmap (1 * minUTxOValue +) currentSrcWalletBalance)
+                rb
+        listAddresses @n ctx wDest >>= verifyAddrs 22 2
```

Comment:
@paweljakubas this doesn't hold true, even after the patch to `knownAddresses`. We send another transaction from the destination wallet back to the source wallet. We should see one more unused address here, or is there simply no change?

Reply:
after sending from wDest we should have external address pool like before it, ie. Used address at index =0, and Unused from 1 up to 10.So 11 addresses in the pool. Now, one of internal addresses is Used. the question is which one. If it is the first one in the internal pool, then the above should be correct expectation, ie. we should have after two pools `<>` 
```
i=0 Used -- beginning of external pool
i=1 Unused
i=2 Unused
...
i=10 Unused -- end of external pool
i=11 Used -- beginning of internal pool
i=12 Unused 
...
i=21 Unused -- end of internal pool
```
It is not like that? maybe it is worth looking at seq state and make sure

Reply:
I think there's just no change generated... I changed the money that is initially sent from `wSrc` to `wDest` to `(3 * minUTxOValue)` and suddenly the test works.

Reply:
yes! I said in the comment above. both output and change NEEDS to be > minUTxOValue and to ensure that let's have bigger input in wDest. So that input - fee = change > minUTxOValue

Reply:
see https://github.com/input-output-hk/cardano-wallet/pull/2370#discussion_r539154549